server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -86,12 +91,48 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, persistErr)
 	}
 
+	// rebase open PRs
+	// worker uses the returned error types to perform some follow up tasks, so instead of propagating the rebase error back, we configure
+	// infinite retries and log the error to ensure we rebase open PRs before deploying another change in this root
+	// ScheduleToCloseTimeout is configured to 8 hours to ensure we're not blocking deploys for this root if the issue persists
+	if rebaseErr := p.rebaseOpenPRsForRoot(ctx, requestedDeployment.Repo); rebaseErr != nil {
+		scope.Counter("rebase_open_prs_err").Inc(1)
+		logger.Error(ctx, "unable to rebase open PRs", key.ErrKey, rebaseErr)
+	}
+
 	// Count this as deployment as latest if it's not a PlanRejectionError which means it is a TerraformClientError
 	// We do this as a safety measure to avoid deploying out of order revision after a failed deploy since it could still
 	// mutate the state file
 	return latestDeployment, err
 }
 
+func (p *Deployer) rebaseOpenPRsForRoot(ctx workflow.Context, repo github.Repo) error {
+	// skip rebasing open PRs if the execution is still on the old workflow version
+	version := workflow.GetVersion(ctx, version.RebaseOpenPRs, workflow.DefaultVersion, 1)
```

Comment:
If you're going to do this, you can't just add part of the implementation because what happens when you add the other activities? I don't think it'll be cleaning to keep adding more versions to support feature development. Here's the how I'd like to see this built:

Your first PR should be the skeleton where all the activities are defined and the logic is written, however, your activities would basically just have empty implementations.  This logic should be tested thoroughly but I don't think we should connect this to the main workflow.  

Then you can implement those activities in separate PRs if you want as follow ups.

Finally you can connect it to the main workflow with a single version check in the last PR and deploy it out.

Reply:
Gotcha, I have followed your suggestion and created skeleton [PR](https://github.com/lyft/atlantis/pull/538) and subsequent PRs to implement the activities.